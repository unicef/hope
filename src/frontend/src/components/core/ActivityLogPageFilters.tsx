import ViewModuleRoundedIcon from '@mui/icons-material/ViewModuleRounded';
import { Grid, MenuItem } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation, useNavigate } from 'react-router-dom';
import { AssigneeAutocomplete } from '@shared/autocompletes/AssigneeAutocomplete';
import { createHandleApplyFilterChange } from '@utils/utils';
import { FiltersSection } from './FiltersSection';
import { SearchTextField } from './SearchTextField';
import { SelectFilter } from './SelectFilter';

interface ActivityLogPageFiltersProps {
  filter;
  setFilter: (filter) => void;
  initialFilter;
  appliedFilter;
  setAppliedFilter: (filter) => void;
}
export function ActivityLogPageFilters({
  filter,
  setFilter,
  initialFilter,
  appliedFilter,
  setAppliedFilter,
}: ActivityLogPageFiltersProps): React.ReactElement {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const location = useLocation();

  const { handleFilterChange, applyFilterChanges, clearFilter } =
    createHandleApplyFilterChange(
      initialFilter,
      navigate,
      location,
      filter,
      setFilter,
      appliedFilter,
      setAppliedFilter,
    );

  const handleApplyFilter = (): void => {
    applyFilterChanges();
  };

  const handleClearFilter = (): void => {
    clearFilter();
  };

  const modules = {
    program: 'Programme',
    household: 'Household',
    individual: 'Individual',
    grievanceticket: 'Grievance Tickets',
    paymentverificationplan: 'Payment Plan Payment Verification',
    targetpopulation: 'Target Population',
    registrationdataimport: 'Registration Data Import',
  };
  return (
    <FiltersSection
      clearHandler={handleClearFilter}
      applyHandler={handleApplyFilter}
    >
      <Grid container alignItems="center" spacing={3}>
        <Grid item xs={3}>
          <SearchTextField
            label={t('Search')}
            value={filter.search}
            onChange={(e) => handleFilterChange('search', e.target.value)}
            data-cy="filters-search"
          />
        </Grid>
        <Grid item xs={3}>
          <SelectFilter
            onChange={(e) => handleFilterChange('module', e.target.value)}
            label={t('Module')}
            value={filter.module}
            icon={<ViewModuleRoundedIcon />}
            SelectDisplayProps={{
              'data-cy': 'filters-residence-status',
            }}
            MenuProps={{
              'data-cy': 'filters-residence-status-options',
            }}
          >
            {Object.entries(modules)
              .sort()
              .map(([key, value]) => (
                <MenuItem key={key} value={key}>
                  {value}
                </MenuItem>
              ))}
          </SelectFilter>
        </Grid>
        <Grid item xs={3}>
          <AssigneeAutocomplete
            label={t('User')}
            filter={filter}
            name="userId"
            value={filter.userId}
            setFilter={setFilter}
            initialFilter={initialFilter}
            appliedFilter={appliedFilter}
            setAppliedFilter={setAppliedFilter}
          />
        </Grid>
      </Grid>
    </FiltersSection>
  );
}
