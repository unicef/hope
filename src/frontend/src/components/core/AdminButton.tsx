import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
import * as React from 'react';


interface AdminButtonProps {
    adminUrl: string;
    [key: string]: any;
}

export const AdminButton: React.FC<AdminButtonProps> = ({
    adminUrl,
    ...otherProps
}) => {
    if (adminUrl) {
        return <a href={adminUrl}><ArrowCircleRightIcon color="primary" {...otherProps}/></a>;
    }
    return null;
};
