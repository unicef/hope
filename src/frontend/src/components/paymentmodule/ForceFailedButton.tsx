import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogContainer } from '@containers/dialogs/DialogContainer';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { useMarkPayAsFailedMutation } from '@generated/graphql';

export interface ForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function ForceFailedButton({
  paymentId,
  disabled = false,
}: ForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useMarkPayAsFailedMutation();

  const submit = async (): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
        },
      });
      setOpenModal(false);
      showMessage(t('Payment has been marked as failed.'));
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-mark-as-failed"
          disabled={disabled}
        >
          {t('Mark as failed')}
        </Button>
      </Box>
      <Dialog
        open={isOpenModal}
        onClose={() => setOpenModal(false)}
        scroll="paper"
        aria-labelledby="form-dialog-title"
        maxWidth="md"
      >
        <DialogTitleWrapper>
          <DialogTitle>{t('Mark as failed')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t('Are you sure you would like to mark payment as failed?')}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setOpenModal(false)}>{t('CANCEL')}</Button>
            <Button
              type="submit"
              color="primary"
              variant="contained"
              onClick={() => submit()}
              data-cy="button-submit"
              disabled={loading}
            >
              {t('Mark as failed')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </Box>
  );
}
