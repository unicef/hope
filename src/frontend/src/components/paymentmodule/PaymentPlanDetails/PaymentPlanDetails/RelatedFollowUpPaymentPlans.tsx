import * as React from 'react';
import { useState } from 'react';
import Button from '@mui/material/Button';
import { Box } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { PaymentPlanQuery } from '@generated/graphql';
import { BlackLink } from '@core/BlackLink';
import { LabelizedField } from '@core/LabelizedField';

interface RelatedFollowUpPaymentPlansProps {
  baseUrl: string;
  followUps: PaymentPlanQuery['paymentPlan']['followUps'];
}

export function RelatedFollowUpPaymentPlans({
  followUps,
  baseUrl,
}: RelatedFollowUpPaymentPlansProps): React.ReactElement {
  const { t } = useTranslation();
  const [showAll, setShowAll] = useState(false);

  const handleButtonClick = (): void => {
    setShowAll(!showAll);
  };

  let followUpLinks = null;
  if (followUps?.edges?.length > 0) {
    const truncatedFollowUps = showAll
      ? followUps.edges
      : followUps.edges.slice(0, 5);
    followUpLinks = truncatedFollowUps.map((followUp) => (
      <BlackLink
        key={followUp?.node?.id}
        to={`/${baseUrl}/payment-module/followup-payment-plans/${followUp?.node?.id}`}
      >
        {followUp?.node?.unicefId}
        <br />
      </BlackLink>
    ));
  }

  return (
    <LabelizedField label={t('Related Follow-Up Payment Plans')}>
      <Box display="flex" flexDirection="column">
        {followUpLinks || '-'}
        {followUps?.edges?.length > 5 && (
          <Button
            variant="outlined"
            color="primary"
            onClick={handleButtonClick}
          >
            {showAll ? t('Hide') : t('See all')}
          </Button>
        )}
      </Box>
    </LabelizedField>
  );
}
