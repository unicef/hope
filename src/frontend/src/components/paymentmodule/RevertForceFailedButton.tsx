import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Field, Form, Formik } from 'formik';
import * as Yup from 'yup';
import CalendarTodayRoundedIcon from '@mui/icons-material/CalendarTodayRounded';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { FormikTextField } from '@shared/Formik/FormikTextField';
import { FormikDateField } from '@shared/Formik/FormikDateField';
import { useRevertMarkPayAsFailedMutation } from '@generated/graphql';

export interface RevertForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function RevertForceFailedButton({
  paymentId,
  disabled = false,
}: RevertForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useRevertMarkPayAsFailedMutation();

  const validationSchema = Yup.object().shape({
    deliveredQuantity: Yup.number()
      .min(0)
      .max(99999999, t('Number is too big')),
    deliveryDate: Yup.date().required(t('Delivery date is required')),
  });

  const submit = async (values, { resetForm }): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
          deliveredQuantity: values.deliveredQuantity,
          deliveryDate: values.deliveryDate,
        },
      });
      setOpenModal(false);
      showMessage(t('Force failed has been reverted.'));
      resetForm();
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-revert-mark-as-failed"
          disabled={disabled}
        >
          {t('Revert mark as failed')}
        </Button>
      </Box>
      <Formik
        initialValues={{
          deliveredQuantity: 0,
          deliveryDate: '',
        }}
        validationSchema={validationSchema}
        onSubmit={submit}
      >
        {({ submitForm, resetForm }) => (
          <Dialog
            open={isOpenModal}
            onClose={() => setOpenModal(false)}
            scroll="paper"
            aria-labelledby="form-dialog-title"
            maxWidth="md"
          >
            <DialogTitleWrapper>
              <DialogTitle>{t('Revert mark as failed')}</DialogTitle>
            </DialogTitleWrapper>
            <DialogContent>
              <Form>
                <Field
                  name="deliveredQuantity"
                  label={t('Delivered Quantity')}
                  type="number"
                  required
                  fullWidth
                  variant="outlined"
                  component={FormikTextField}
                />
                <Field
                  name="deliveryDate"
                  label={t('Delivery Date')}
                  component={FormikDateField}
                  required
                  fullWidth
                  decoratorEnd={<CalendarTodayRoundedIcon color="disabled" />}
                />
              </Form>
            </DialogContent>
            <DialogFooter>
              <DialogActions>
                <Button
                  onClick={() => {
                    setOpenModal(false);
                    resetForm();
                  }}
                >
                  {t('CANCEL')}
                </Button>
                <Button
                  type="submit"
                  color="primary"
                  variant="contained"
                  onClick={submitForm}
                  data-cy="button-submit"
                  disabled={loading}
                >
                  {t('Revert mark as failed')}
                </Button>
              </DialogActions>
            </DialogFooter>
          </Dialog>
        )}
      </Formik>
    </Box>
  );
}
