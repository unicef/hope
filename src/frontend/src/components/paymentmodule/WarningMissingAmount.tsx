import { Box } from '@mui/material';
import WarningIcon from '@mui/icons-material/Warning';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { ReactElement } from 'react';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): ReactElement {
  const { t } = useTranslation();

  return (
    <Box display="flex" alignItems="center">
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
