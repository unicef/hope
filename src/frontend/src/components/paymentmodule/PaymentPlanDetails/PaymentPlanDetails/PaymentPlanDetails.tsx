import {
  Box,
  Grid2 as Grid,
  IconButton,
  Tooltip,
  Typography,
} from '@mui/material';
import { useTranslation } from 'react-i18next';
import { renderUserName } from '@utils/utils';
import { BlackLink } from '@core/BlackLink';
import { ContainerColumnWithBorder } from '@core/ContainerColumnWithBorder';
import { LabelizedField } from '@core/LabelizedField';
import { OverviewContainer } from '@core/OverviewContainer';
import { Title } from '@core/Title';
import { UniversalMoment } from '@core/UniversalMoment';
import { FieldBorder } from '@core/FieldBorder';
import { RelatedFollowUpPaymentPlans } from './RelatedFollowUpPaymentPlans';
import { Info } from '@mui/icons-material';
import { ReactElement } from 'react';
import withErrorBoundary from '@components/core/withErrorBoundary';
import { PaymentPlanDetail } from '@restgenerated/models/PaymentPlanDetail';

interface PaymentPlanDetailsProps {
  baseUrl: string;
  paymentPlan: PaymentPlanDetail;
}

const PaymentPlanDetails = ({
  baseUrl,
  paymentPlan,
}: PaymentPlanDetailsProps): ReactElement => {
  const { t } = useTranslation();
  const {
    created_by,
    program,
    currency,
    start_date,
    end_date,
    exchange_rate,
    dispersion_start_date,
    dispersion_end_date,
    follow_ups,
  } = paymentPlan;

  return (
    <Grid size={{ xs: 12 }}>
      <ContainerColumnWithBorder>
        <Title>
          <Typography variant="h6">{t('Details')}</Typography>
        </Title>
        <OverviewContainer>
          <Grid container>
            <Grid container size={{ xs: 9 }} spacing={6}>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Created By')}>
                  {renderUserName(created_by)}
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Programme')}>
                  <BlackLink to={`/${baseUrl}/details/${program.id}`}>
                    {program.name}
                  </BlackLink>
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Target Population')}>
                  <BlackLink
                    to={`/${baseUrl}/target-population/${paymentPlan.id}`}
                  >
                    {paymentPlan.name}
                  </BlackLink>
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Currency')}>
                  {currency}
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Start Date')}>
                  <UniversalMoment>{startDate}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('End Date')}>
                  <UniversalMoment>{endDate}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Dispersion Start Date')}>
                  <UniversalMoment>{dispersion_start_date}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <LabelizedField label={t('Dispersion End Date')}>
                  <UniversalMoment>{dispersion_end_date}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid size={{ xs: 3 }}>
                <Box mr={1}>
                  <LabelizedField label={t('FX Rate Applied')}>
                    {exchange_rate}
                  </LabelizedField>
                </Box>
                <Tooltip
                  title={t(
                    'If displayed exchange rate differs from Vision, please contact your designated focal point for resolution',
                  )}
                >
                  <IconButton
                    color="primary"
                    aria-label="exchange-rate"
                    data-cy="info-exchange-rate"
                  >
                    <Info />
                  </IconButton>
                </Tooltip>
              </Grid>
            </Grid>
            <Grid container direction="column" size={{ xs: 3 }} spacing={6}>
              <Grid size={{ xs: 12 }}>
                <FieldBorder color="#84A1CA">
                  <RelatedFollowUpPaymentPlans
                    followUps={follow_ups}
                    baseUrl={baseUrl}
                  />
                </FieldBorder>
              </Grid>
            </Grid>
          </Grid>
        </OverviewContainer>
      </ContainerColumnWithBorder>
    </Grid>
  );
};

export default withErrorBoundary(PaymentPlanDetails, 'PaymentPlanDetails');
