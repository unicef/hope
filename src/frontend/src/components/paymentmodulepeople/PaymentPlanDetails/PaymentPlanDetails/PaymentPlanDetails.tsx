import { Box, Grid, IconButton, Tooltip, Typography } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { renderUserName } from '@utils/utils';
import { PaymentPlanQuery } from '@generated/graphql';
import { BlackLink } from '@core/BlackLink';
import { ContainerColumnWithBorder } from '@core/ContainerColumnWithBorder';
import { LabelizedField } from '@core/LabelizedField';
import { OverviewContainer } from '@core/OverviewContainer';
import { Title } from '@core/Title';
import { UniversalMoment } from '@core/UniversalMoment';
import { FieldBorder } from '@core/FieldBorder';
import { RelatedFollowUpPaymentPlans } from './RelatedFollowUpPaymentPlans';
import { Info } from '@mui/icons-material';

interface PaymentPlanDetailsProps {
  baseUrl: string;
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const PaymentPlanDetails = ({
  baseUrl,
  paymentPlan,
}: PaymentPlanDetailsProps): React.ReactElement => {
  const { t } = useTranslation();
  const {
    createdBy,
    program,
    targetPopulation,
    currency,
    startDate,
    endDate,
    dispersionStartDate,
    dispersionEndDate,
    followUps,
    exchangeRate,
  } = paymentPlan;

  return (
    <Grid item xs={12}>
      <ContainerColumnWithBorder>
        <Title>
          <Typography variant="h6">{t('Details')}</Typography>
        </Title>
        <OverviewContainer>
          <Grid container>
            <Grid container item xs={9} spacing={6}>
              <Grid item xs={3}>
                <LabelizedField label={t('Created By')}>
                  {renderUserName(createdBy)}
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('Programme')}>
                  <BlackLink to={`/${baseUrl}/details/${program.id}`}>
                    {program.name}
                  </BlackLink>
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('Target Population')}>
                  <BlackLink
                    to={`/${baseUrl}/target-population/${targetPopulation.id}`}
                  >
                    {targetPopulation.name}
                  </BlackLink>
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('Currency')}>
                  {currency}
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('Start Date')}>
                  <UniversalMoment>{startDate}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('End Date')}>
                  <UniversalMoment>{endDate}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('Dispersion Start Date')}>
                  <UniversalMoment>{dispersionStartDate}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <LabelizedField label={t('Dispersion End Date')}>
                  <UniversalMoment>{dispersionEndDate}</UniversalMoment>
                </LabelizedField>
              </Grid>
              <Grid item xs={3}>
                <Box mr={1}>
                  <LabelizedField label={t('FX Rate Applied')}>
                    {exchangeRate}
                  </LabelizedField>
                </Box>
                <Tooltip
                  title={t(
                    'If displayed exchange rate differs from Vision, please contact your designated focal point for resolution',
                  )}
                >
                  <IconButton
                    color="primary"
                    aria-label="exchange-rate"
                    data-cy="info-exchange-rate"
                  >
                    <Info />
                  </IconButton>
                </Tooltip>
              </Grid>
            </Grid>
            <Grid container direction="column" item xs={3} spacing={6}>
              <Grid item xs={12}>
                <FieldBorder color="#84A1CA">
                  <RelatedFollowUpPaymentPlans
                    followUps={followUps}
                    baseUrl={baseUrl}
                  />
                </FieldBorder>
              </Grid>
            </Grid>
          </Grid>
        </OverviewContainer>
      </ContainerColumnWithBorder>
    </Grid>
  );
};
