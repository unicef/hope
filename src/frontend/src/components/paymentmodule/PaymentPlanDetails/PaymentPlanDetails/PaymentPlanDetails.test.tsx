import { fakeApolloPaymentPlan } from '../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { render } from '../../../../testUtils/testUtils';
import { fakeBaseUrl } from '../../../../../fixtures/core/fakeBaseUrl';
import PaymentPlanDetails from './PaymentPlanDetails';

describe('components/paymentmodule/PaymentPlanDetails/PaymentPlanDetails', () => {
  it('should render', () => {
    const { container } = render(
      <PaymentPlanDetails
        baseUrl={fakeBaseUrl}
        paymentPlan={fakeApolloPaymentPlan}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
