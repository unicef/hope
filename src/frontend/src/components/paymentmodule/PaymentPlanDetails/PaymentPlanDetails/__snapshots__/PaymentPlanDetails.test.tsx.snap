// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/PaymentPlanDetails should render 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-dVBluf ebFzuo css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="sc-gNZgCX kFsGVB"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-66ugg1-MuiTypography-root"
        >
          Details
        </h6>
      </div>
      <div
        class="sc-jbAkgO jpddxj"
      >
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1jxwjqx-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-9 MuiGrid2-spacing-xs-6 css-kel0za-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Created By
                </span>
                <div
                  data-cy="label-Created By"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    Matthew Sosa
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Programme
                </span>
                <div
                  data-cy="label-Programme"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    <a
                      class="sc-FEMpB gVfMNY"
                      href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/details/UHJvZ3JhbU5vZGU6NWJhMjEzY2UtNmNlOS00NTc4LThhNDgtYjFmMDgyM2Q2MDAy"
                    >
                      Already attention fear well hit instead person.
                    </a>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Target Population
                </span>
                <div
                  data-cy="label-Target Population"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    <a
                      class="sc-FEMpB gVfMNY"
                      href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/target-population/UGF5bWVudFBsYW5Ob2RlOmE5YzJjMmM4LWJmYWUtNDBhMy05YmYwLWIxYWE1ZmRlMDE0YQ=="
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Currency
                </span>
                <div
                  data-cy="label-Currency"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    PLN
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Start Date
                </span>
                <div
                  data-cy="label-Start Date"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    <time
                      datetime="1603756800000"
                    >
                      27 Oct 2020
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  End Date
                </span>
                <div
                  data-cy="label-End Date"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    <time
                      datetime="1631059200000"
                    >
                      8 Sep 2021
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Dispersion Start Date
                </span>
                <div
                  data-cy="label-Dispersion Start Date"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    <time
                      datetime="1842307200000"
                    >
                      19 May 2028
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div>
                <span
                  class="sc-fLDLck ivXKQK"
                  color="textSecondary"
                >
                  Dispersion End Date
                </span>
                <div
                  data-cy="label-Dispersion End Date"
                >
                  <span
                    class="sc-dycHyt kguSQI"
                    color="textSecondary"
                  >
                    <time
                      datetime="1880755200000"
                    >
                      7 Aug 2029
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-3 css-hxbzlh-MuiGrid2-root"
            >
              <div
                class="MuiBox-root css-15ro776"
              >
                <div>
                  <span
                    class="sc-fLDLck ivXKQK"
                    color="textSecondary"
                  >
                    FX Rate Applied
                  </span>
                  <div
                    data-cy="label-FX Rate Applied"
                  >
                    -
                  </div>
                </div>
              </div>
              <button
                aria-label="exchange-rate"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-9mgek4-MuiButtonBase-root-MuiIconButton-root"
                data-cy="info-exchange-rate"
                data-mui-internal-clone-element="true"
                id=":r1:"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-column MuiGrid2-grid-xs-3 MuiGrid2-spacing-xs-6 css-gxd0v3-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="sc-kfeOyU fMDCaE"
                color="#84A1CA"
              >
                <div>
                  <span
                    class="sc-fLDLck ivXKQK"
                    color="textSecondary"
                  >
                    Related Follow-Up Payment Plans
                  </span>
                  <div
                    data-cy="label-Related Follow-Up Payment Plans"
                  >
                    <span
                      class="sc-dycHyt kguSQI"
                      color="textSecondary"
                    >
                      <div
                        class="MuiBox-root css-j7qwjs"
                      >
                        -
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
